daemon/image_pull.go

```diff
@@ -113,10 +112,6 @@ func (daemon *Daemon) GetRepository(ctx context.Context, ref reference.Named, au
 	)
 
 	for _, endpoint := range endpoints {
-		if endpoint.Version == registry.APIVersion1 {
-			continue
```

Comment:
Instead of removing this; should we produce an error here? Although the codebase in this repository should no longer return v1 endpoints, the code is just _one_ implementation of the interface; other implementations may still return v1 endpoints (or mixed v1/v2 endpoints); https://github.com/moby/moby/blob/master/registry/service.go#L28-L29

Reply:
We could remove `registry.APIVersion1` so that it could no longer be used by other implementations